### PR TITLE
chore(filter2): test updates

### PR DESCRIPTION
**Pull request by siphiuel**
# Description
Includes a refactor of filter v2 tests, plus some minor changes taken from js-waku (https://github.com/waku-org/js-waku/blob/master/packages/tests/tests/filter.node.spec.ts)

Many changed files are due to function renamings done for consistency - not sure if this was the right call.

### CODE REVIEW

File: waku/v2/protocol/filter/server.go [hunk 1]
```diff
@@ -69,6 +69,10 @@ func (wf *WakuFilterFullNode) SetHost(h host.Host) {
 	wf.h = h
 }
 
+func (wf *WakuFilterFullNode) Host() host.Host {
```
Comment (richard-ramos) [inline waku/v2/protocol/filter/server.go:72]: The `Host` function in here, lightnode and relay aren't necessary as you would have access to the host via the WakuNode.

**Review by richard-ramos — COMMENTED** [review]
Thank you for the PR. Left some minor comments:

**Comment by richard-ramos** [comment]
@siphiuel, lint and build steps are failing.

File: waku/v2/protocol/filter/server.go (hunk 1 above)
Comment (siphiuel) [reply waku/v2/protocol/filter/server.go]: Understood, done.

**Comment by siphiuel** [comment]
@richard-ramos fixed lint and build

**Review by richard-ramos — APPROVED** [review]
:rocket:

**Pull request merged**